packages/grid/_modules_/grid/hooks/features/pagination/useGridPagination.ts

```diff
@@ -6,94 +6,121 @@ import { GridPageChangeParams } from '../../../models/params/gridPageChangeParam
 import { useGridApiOptionHandler } from '../../root/useGridApiEventHandler';
 import { useGridApiMethod } from '../../root/useGridApiMethod';
 import { optionsSelector } from '../../utils/optionsSelector';
+import { useGridState } from '../core/useGridState';
 import { gridContainerSizesSelector } from '../../root/gridContainerSizesSelector';
 import { useLogger } from '../../utils/useLogger';
-import { useGridReducer } from '../core/useGridReducer';
 import { useGridSelector } from '../core/useGridSelector';
 import { visibleGridRowCountSelector } from '../filter/gridFilterSelector';
-import {
-  GRID_INITIAL_PAGINATION_STATE,
-  GridPaginationActions,
-  gridPaginationReducer,
-  GridPaginationState,
-  setGridPageActionCreator,
-  setGridPageSizeActionCreator,
-  setGridPaginationModeActionCreator,
-  setGridRowCountActionCreator,
-} from './gridPaginationReducer';
+import { GridPaginationState } from './gridPaginationState';
 
-const PAGINATION_STATE_ID = 'pagination';
+function computeState(state) {
```

Comment:
I would also rename `computeState` to `getUpdatedState`

Reply:
I have updated the name of the function to better describe the use case. It's meant to compute derivative states. It could be done inside a selector but 🤷‍♂️